packages/nx/src/daemon/server/plugins.ts

```diff
@@ -1,18 +1,33 @@
+import { hashObject } from '../../hasher/file-hasher';
 import { readNxJson } from '../../config/nx-json';
 import {
   LoadedNxPlugin,
   loadNxPlugins,
 } from '../../project-graph/plugins/internal-api';
 import { workspaceRoot } from '../../utils/workspace-root';
 
+let currentPluginsConfigurationHash: string;
 let loadedPlugins: LoadedNxPlugin[];
 let cleanup: () => void;
 
 export async function getPlugins() {
-  if (loadedPlugins) {
+  const pluginsConfiguration = readNxJson().plugins ?? [];
+  const pluginsConfigurationHash = hashObject(pluginsConfiguration);
+
+  // If the plugins configuration has not changed, reuse the current plugins
+  if (
+    loadedPlugins &&
+    pluginsConfigurationHash === currentPluginsConfigurationHash
+  ) {
     return loadedPlugins;
   }
-  const pluginsConfiguration = readNxJson().plugins ?? [];
+
+  // Cleanup current plugins before loading new ones
+  if (cleanup) {
+    cleanup();
+  }
```

Comment:
Currently isolated plugins don't actually return cleanup functions. I'll fix that.